src/hotspot/share/logging/logTagSet.cpp

```diff
@@ -126,12 +126,17 @@ void LogTagSet::vwrite(LogLevelType level, const char* fmt, va_list args) {
   assert(ret >= 0, "Log message buffer issue");
   if ((size_t)ret >= sizeof(buf)) {
     size_t newbuf_len = prefix_len + ret + 1;
-    char* newbuf = NEW_C_HEAP_ARRAY(char, newbuf_len, mtLogging);
+    char* newbuf = (char*)::calloc(newbuf_len, sizeof(char));
```

Comment:
Do we need to zero the array with calloc?
The `(char*)` cast shouldn't be needed.

Reply:
The calloc query has not been addressed.

Reply:
malloc/calloc return void* so need this cast.

Reply:
calloc zero-fills the buffer. See https://en.cppreference.com/w/c/memory/calloc